src/components/LanguageProvider/LanguageProvider.tsx

```diff
@@ -0,0 +1,27 @@
+import { ReactNode, FC, useMemo, useState } from 'react';
+import Languages from 'constants/languages';
+import LanguageContext from 'contexts/LanguageContext';
+import LanguageContextType from 'types/languageContextType';
+
+interface Props {
+  children: ReactNode;
+}
+
+const LanguageProvider: FC<Props> = function LanguageProvider({ children }) {
+  const [selectedLanguage, onSelectLanguage] = useState<Languages>(
+    Languages.ENGLISH,
+  );
+
+  const language: LanguageContextType = useMemo(
```

Comment:
You don't need it. Let's discuss why this memoization won't work.
If you want to ensure better performance, wrap `onSelectLanguage` in `useCallback`